### ISSUE TITLE
`~/.config/environment.d` conf files are not loaded in cosmic-session

### ISSUE THREAD

**Issue by ryanabx** [pop-os/cosmic-session#76]
Version:

`cosmic-session-1.0.0~alpha.2^git20241016.c3a35f3-1.fc41.x86_64`

systemd environment files can be set in these places:

System-Wide Directories:

    /etc/environment.d/
    /run/environment.d/
    /usr/lib/environment.d/

User-Specific Directory:

    ~/.config/environment.d/

I put an environment file in `~/.config/environment.d` and it wasn't loaded by cosmic-session. kde's session loads it.

**Comment by Drakulix** [comment]
I am pretty sure, that would be a configuration issue (though I would be very curious to know what KDE does different here exactly) as all cosmic-session is doing is launching a login-shell. Loading profile and environment should be handled by that?

**Comment by wiiznokes** [comment]
I ran is this issue for my applet https://github.com/cosmic-utils/clipboard-manager/pull/82#issue-2549413499

**Comment by ryanabx** [comment]
> I am pretty sure, that would be a configuration issue (though I would be very curious to know what KDE does different here exactly) as all cosmic-session is doing is launching a login-shell. Loading profile and environment should be handled by that?

Is this reproducible on Pop? If not it definitely could be a fedora only issue.

**Comment by Vixea** [comment]
Reproducable in serpent-os, its caused by a lack of explicitly getting environment variables it seems.

**Comment by Vixea** [comment]
A person working with/on serpentos has made a patch
https://github.com/serpent-os/recipes/pull/507
They'll submit it eventually here but if your affected by it and want it working now just apply the patch

**Comment by ReillyBrogan** [comment]
Yes I created a quick and dirty patch there since we needed to propagate certain variable from the user manager into the session. If [the patch](https://github.com/serpent-os/recipes/blob/main/c/cosmic-session/pkg/0001-systemd-env.patch) looks reasonable as-is I can clean it up a bit for a PR but realistically I'm not sure I'll have time if more work is requested than that.

For reference I largely adapted the [code from plasma-workspace](https://invent.kde.org/plasma/plasma-workspace/-/blob/Plasma/6.2/startkde/startplasma.cpp?ref_type=heads#L477) which does the same thing. Both Plasma and GNOME directly support importing the user environment from systemd in case the login manager doesn't support doing so before the session is started (I think only GDM does that currently). I didn't really spend any time tracking down the implementation for GNOME once I had plasma-workspace in front of me as a reference however.
